skimage/io/collection.py

```diff
@@ -147,21 +144,22 @@ def imread_convert(f):
     >>> coll[0].shape
     (200, 200)
 
-    >>> ic = io.ImageCollection('/tmp/work/*.png:/tmp/other/*.jpg')
-
+    >>> ic = io.ImageCollection(['/tmp/work/*.png', '/tmp/other/*.jpg'])
     """
 
     def __init__(self, load_pattern, conserve_memory=True, load_func=None,
                  **load_func_kwargs):
         """Load and manage a collection of images."""
-        if isinstance(load_pattern, str):
+        try:
+            if not isinstance(load_pattern, str):
```

Comment:
I think this check should be more stringent: it should not be a string, but it should be a sequence.

Also, `try/except` clauses should contain as little content as possible. I think this one encompasses too many statements.

Reply:
Tried to fix that with:
```
if isinstance(load_pattern, (list, tuple)):
```
IMHO this `try/except` is as good as it gets. It receives the three types of inputs we'd like to put there, and it's rather efficient when compared to an `if isinstance(load_pattern, (str, list, tuple)`. That could also bring us issues in a more difficult type of input:
```
load_pattern = [(1, 'one'), (2, 'two')]
In []: isinstance(, (list, tuple))                     
Out[]: True
```